pandas/io/parsers.py

```diff
@@ -2165,6 +2231,129 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
         return values
 
 
+class BytesIOWrapper:
+    """
+    Allows the pyarrow engine for read_csv() to read from string buffers
+    """
+
+    def __init__(
+        self,
+        string_buffer: Union[StringIO, TextIOBase],
+        encoding: str = "utf-8",
+    ):
+        self.string_buffer = string_buffer
+        self.encoding = encoding
+
+    def __getattr__(self, attr: str):
+        return getattr(self.string_buffer, attr)
+
+    def read(self, size: int = -1):
+        content = self.string_buffer.read(size)
+        return content.encode(self.encoding)
+
+
+class ArrowParserWrapper(ParserBase):
+    """
+    Wrapper for the pyarrow engine for read_csv()
+    """
+
+    def __init__(self, src, **kwds):
+        self.kwds = kwds
+        self.src = src
+
+        ParserBase.__init__(self, kwds)
+
+        self._parse_kwds()
+
+    def _parse_kwds(self):
+        encoding: Optional[str] = self.kwds.get("encoding")
+        self.encoding = "utf-8" if encoding is None else encoding
+
+        self.usecols, self.usecols_dtype = _validate_usecols_arg(self.kwds["usecols"])
+        na_values = self.kwds["na_values"]
+        if isinstance(na_values, dict):
+            raise ValueError(
+                "The pyarrow engine doesn't support passing a dict for na_values"
+            )
+        self.na_values = list(
+            _clean_na_values(
+                self.kwds["na_values"], keep_default_na=self.kwds["keep_default_na"]
+            )[0]
+        )
+
+        if isinstance(self.src, TextIOBase):
+            self.src = BytesIOWrapper(self.src, encoding=self.encoding)
+
+    def _get_pyarrow_options(self):
+        # rename some arguments to pass to pyarrow
+        mapping = {
+            "usecols": "include_columns",
+            "na_values": "null_values",
+            "escapechar": "escape_char",
+            "skip_blank_lines": "ignore_empty_lines",
+        }
+        for pandas_name, pyarrow_name in mapping.items():
+            if pandas_name in self.kwds and self.kwds.get(pandas_name) is not None:
+                self.kwds[pyarrow_name] = self.kwds.pop(pandas_name)
+
+        self.parse_options = {
+            option_name: option_value
+            for option_name, option_value in self.kwds.items()
+            if option_value is not None
+            and option_name
+            in ("delimiter", "quote_char", "escape_char", "ignore_empty_lines")
+        }
+        self.convert_options = {
+            option_name: option_value
+            for option_name, option_value in self.kwds.items()
+            if option_value is not None
+            and option_name
+            in ("include_columns", "null_values", "true_values", "false_values")
+        }
+        self.read_options = {
+            "autogenerate_column_names": self.header is None,
+            "skip_rows": self.header
+            if self.header is not None
+            else self.kwds["skiprows"],
+        }
+
+    def _finalize_output(self, frame):
```

Comment:
it would be good (ideally as a pre-cursor) to make _finalize_output a method on ParserBase and implement it for c/python (instead of some of the code on L1120).

Reply:
Will do